worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/TierWriter.scala

```diff
@@ -643,13 +643,26 @@ class DfsTierWriter(
       hadoopFs.delete(hdfsFileInfo.getDfsPath, false)
       deleted = true
     } else {
-      hadoopFs.create(hdfsFileInfo.getDfsWriterSuccessPath).close()
-      val indexOutputStream = hadoopFs.create(hdfsFileInfo.getDfsIndexPath)
-      indexOutputStream.writeInt(hdfsFileInfo.getReduceFileMeta.getChunkOffsets.size)
-      for (offset <- hdfsFileInfo.getReduceFileMeta.getChunkOffsets.asScala) {
-        indexOutputStream.writeLong(offset)
+      var retryCount = 0
+      while (retryCount < conf.workerCreateIndexMaxAttempts && !hadoopFs.exists(hdfsFileInfo.getDfsWriterSuccessPath)) {
+        try {
+          if (hadoopFs.exists(hdfsFileInfo.getDfsIndexPath)) {
+            hadoopFs.delete(hdfsFileInfo.getDfsIndexPath, true)
+          }
+          val indexOutputStream = hadoopFs.create(hdfsFileInfo.getDfsIndexPath)
+          indexOutputStream.writeInt(hdfsFileInfo.getReduceFileMeta.getChunkOffsets.size)
+          for (offset <- hdfsFileInfo.getReduceFileMeta.getChunkOffsets.asScala) {
+            indexOutputStream.writeLong(offset)
+          }
+          indexOutputStream.close()
+          hadoopFs.create(hdfsFileInfo.getDfsWriterSuccessPath).close()
```

Comment:
If creating the success path results in an IOException, it will lead to unnecessary retries for creating the index file. Therefore, it would be better to implement a retry  for IOException during both the creation of the index file and the success path.